server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -498,6 +496,20 @@ public Repository repository(String repositoryName) {
         throw new RepositoryMissingException(repositoryName);
     }
 
+    /**
+     * Return repository with the given UUID
+     *
+     * @throws RepositoryMissingException if a repository with that UUID is not registered
+     */
+    public Repository repositoryByUuid(String repositoryUuid, String originalName) {
```

Comment:
This is kind of a strange method since its behavior depends on the current state of whether or not the repo was already used or verified in some form. I guess we'll address this in a follow-up, but maybe we should keep this code logically contained in the searchable snapshots directory and just add a getter for `org.elasticsearch.repositories.RepositoriesService#repositories` to the repo service?

Reply:
Sure, that makes sense to me.